projects/dynamicforms/src/components/upload/upload.component.ts

```diff
@@ -85,17 +87,20 @@ export class UploadComponent extends AbstractFormComponent implements OnInit {
           map(formData => JSON.parse(formData.content).relevantFiles),
           switchMap(codes => {
             return this.fileUploadService.getFiles(codes).pipe(
-              filter(files => !!files.documents),
               map(files => {
-                this.fileList = files.documents;
-                if (this.files.length === 0) {
-                  files.documents.forEach(file => {
-                    this.files.push(file.code);
-                  });
-                  this.uploadControl?.setValue(this.files);
+                if (files && files.documents) {
+                  this.fileList = files.documents;
+                  if (this.files.length === 0) {
```

Comment:
I know this isn't your code, but can you please check whether this if is necessary? Seems redundant.